api/capturesource_test.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/ausocean/openfish/api/api"
+	"github.com/ausocean/openfish/api/ds_client"
+	"github.com/ausocean/openfish/api/handlers"
+	"github.com/gofiber/fiber/v2"
+)
+
+func TestCreateCaptureSource(t *testing.T) {
+
+	testcases := []struct {
+		reqBody   string
+		resStatus int
+	}{
+		// TODO: Test cases with missing fields cause the server to crash.
+		//       Add test case for this and fix issue.
+		{
+			resStatus: fiber.StatusOK,
+			reqBody: `{
+				"name": "Ochre Point",
+				"location": "-35.221389,138.46333333",
+				"camera_hardware": "rpi cam v2 - wide angle lens",
+				"site_id": 10192840284
+			}`,
+		},
+		{
+			resStatus: fiber.StatusOK,
+			reqBody: `{
+				"name": "Ochre Point",
+				"location": "-35.221389,138.46333333",
+				"camera_hardware": "rpi cam v2 - wide angle lens"
+			}`,
+		},
+		{
+			resStatus: fiber.StatusOK,
+			reqBody: `{
+				"name": "Ochre Point",
+				"location": "-35.221389,138.46333333"
+			}`,
+		},
+		{
+			resStatus: fiber.StatusBadRequest,
+			reqBody: `{
+				"name": true,
+				"location": [1,2,3],
+				"camera_hardware": 123456,
+				"site_id": false,
+				"extra": "fields"
+			}`,
+		},
+		{
+			resStatus: fiber.StatusBadRequest,
+			reqBody:   "something that is not json",
+		},
+		{
+			resStatus: fiber.StatusBadRequest,
+			reqBody:   "",
+		},
+	}
+
+	// Initialize app.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	for _, testcase := range testcases {
```

Comment:
if grab the index too, you can use it in the failure cases below to indicate which test number failed ?

Reply:
alternatively add a name field to the anonymous test case struct so that you can name each test.

Reply:
I have added index here and in failure messages